src/util/container.h

```diff
@@ -376,21 +376,33 @@ class ModifySafeMap
 			assert(false);
 			return;
 		}
-		if (m_iterating)
-			m_new.emplace(key, value);
-		else
-			m_values.emplace(key, value);
+		if (m_iterating) {
+			auto it = m_values.find(key);
+			if (it != m_values.end()) {
+				it->second = V();
```

Comment:
Can't we do just
```cpp
it->second = value;
return;
```
here?

Reply:
That's possible and works. But in the context of our usecase I think it makes sense that all kind of new values only show up after the iteration is finished.

Reply:
Makes sense